e2e/cypress/integration/api/iam/01_projects_api.spec.ts

```diff
@@ -0,0 +1,166 @@
+describe('projects API', () => {
+  describe('applying project rules', () => {
+
+    const admin_token = 'ZNhnqJyDqgkXVk_bpobS3YhIMz0='
+
+    const avengers_project = {
+      id: `avengers-project-${Cypress.moment().format('MMDDYYhhmm')}`,
+      name: "Test Avengers Project"
+    }
+
+    const xmen_project = {
+      id: `xmen-project-${Cypress.moment().format('MMDDYYhhmm')}`,
+      name: "Test X-men Project"
+    }
+
+    const avengers_rule = {
+      id: "avengers-rule-1",
+      name: "first rule of avengers project",
+      type: "NODE",
+      project_id: avengers_project.id,
+      conditions: [
+        {
+          attribute: "CHEF_ORGS",
+          operator: "EQUALS",
+          values: ["avengers"]
+        }
+      ]
+    }
+
+    const xmen_rule = {
+      id: "xmen-rule-1",
+      name: "first rule of xmen project",
+      type: "NODE",
+      project_id: xmen_project.id,
+      conditions: [
+        {
+          attribute: "CHEF_ORGS",
+          operator: "EQUALS",
+          values: ["xmen"]
+        }
+      ]
+    }
+
+    before(() => {
+      cy.adminLogin('/').then(() => {
+        let admin = JSON.parse(<string>localStorage.getItem('chef-automate-user'))
+
+        // generate admin API token
+        let token_id = `test-token-${Cypress.moment().format('MMDDYYhhmm')}`
+        cy.request({
+          auth: { bearer: admin.id_token },
+          method: 'POST',
+          url: '/apis/iam/v2beta/tokens',
+          failOnStatusCode: false,
+          body: {
+            id: token_id,
+            name: "cypress-api-test-admin-token",
+            value: admin_token
+          }
+        }).then((response) => {
+          expect([200, 409]).to.include(response.status)
+        })
+
+        cy.request({
+          auth: { bearer: admin.id_token },
+          method: 'POST',
+          url: '/apis/iam/v2beta/policies/administrator-access/members:add',
+          body: {
+            members: [`token:${token_id}`]
+          }
+        })
+
+        // TODO fix this type
+        const nodes = <any>[]
+
+        Cypress.Promise.all([
```

Comment:
decided to go with [this suggestion](https://github.com/cypress-io/cypress/issues/2932#issuecomment-476608299) over cypress' recommendation

Reply:
turned out this didn't work so lots of callbacks it is